packages/api/src/external/aws/address.ts

```diff
@@ -1,39 +1,82 @@
-import { Address } from "../../domain/medical/address";
 import {
-  getCoordinatesFromLocation,
+  AddressSuggestion,
+  parseSuggestedAddress,
+  getLocationResultPayload,
   makeLocationClient,
 } from "@metriport/core/external/aws/location";
-import { Config } from "../../shared/config";
 import * as AWS from "aws-sdk";
-import { Coordinates } from "@metriport/core/external/aws/location";
+import { Address } from "../../domain/medical/address";
+import { Config } from "../../shared/config";
+
+export type AddressAndSuggestedLabel = {
+  address: Address;
+  relevance: number;
+  suggestedLabel: string;
+};
 
 const indexName = Config.getPlaceIndexName();
 const placeIndexRegion = Config.getPlaceIndexRegion();
 const client = makeLocationClient(placeIndexRegion);
 
+export function buildAddressText(address: Address): string {
+  return `${address.addressLine1}, ${address.city}, ${address.state} ${address.zip}`;
+}
+
 /**
- * Geocodes a list of addresses using Amazon Location Services.
- * @param addresses
- * @returns
+ * Adds coordinates to addresses that don't already have them.
+ *
+ * @param addressList a list of Address objects.
+ * @returns a list of updated Address objects and their relevance score from geocoding.
  */
-export async function geocodeAddresses(addresses: Address[]): Promise<Coordinates[]> {
-  const resultPromises = await Promise.allSettled(
-    addresses.map(async address => {
-      const addressText = `${address.addressLine1}, ${address.city}, ${address.state} ${address.zip}`;
-      const countryFilter = address.country ?? "USA";
-
-      const params: AWS.Location.Types.SearchPlaceIndexForTextRequest = {
-        Text: addressText,
-        MaxResults: 1,
-        Language: "en",
-        FilterCountries: [countryFilter],
-        IndexName: indexName,
-      };
-
-      const locationResponse = await client.searchPlaceIndexForText(params).promise();
-      return getCoordinatesFromLocation({ result: locationResponse });
+export async function addGeographicCoordinates(
```

Comment:
Correct me if I'm wrong, but this function is more command/domain than external/tech-specific, right?

I mean, it has logic that's not related specifically to AWS - it uses AWS-specific code `geocodeAddress`, but that's it.

So it could, in theory, be moved to a command?